src/content/1.7/modules/concepts/templating/vuejs/_index.md

```diff
@@ -0,0 +1,206 @@
+
+Create or edit `_dev/vue.config.js` with:
+
+```js
+const path = require('path');
+
+module.exports = {
+  chainWebpack: (config) => {
+    // Stop generating the HTML page
+    config.plugins.delete('html');
+    config.plugins.delete('preload');
+    config.plugins.delete('prefetch');
+
+    // Allow resolving images in the subfolder src/assets/ 
+    config.resolve.alias.set('@', path.resolve(__dirname, 'src'));
+  },
+  css: {
+    extract: false,
+  },
+  runtimeCompiler: true,
+  productionSourceMap: false,
+  filenameHashing: false,
+  // These rules allow the files to be compiled and stored in the proper folder
+  outputDir: '../views/',
+  assetsDir: '',
+  // ⚠️ Change this line with your module name
+  publicPath: '../modules/<module_name>/views/',
+};
+```
+
+{{% notice warning %}}
+This configuration will delete all files present in the `views/js/` folder. Before going further,
+make sure:
+
+* there are no existing files that would be lost, or
+* to write your generated files in an additional subfolder inside `views/js/` by modifying
+`outputDir` & `publicPath`.
+{{% /notice %}}
+
+Then, the scripts available in the package.json can be replaced by these ones:
+
+*modules/[module_name]/_dev/package.json*
+
+```js
+    "build": "vue-cli-service build --no-clean",
+    "lint": "vue-cli-service lint --fix",
+    "dev": "vue-cli-service build --no-clean --mode development --watch",
+```
+
+The new `dev` script allow to compile the JS files, and keep watching at any change that would happen
+in the project directories. Everytime one file is saved, the compilation will run again immediately.
+
+Testing the compilation of the files is done with:
+
+```
+$ npm run dev
+```
+
+After some time, a similar output should appear on the terminal once done:
+
+```
+$ npm run dev
+
+[...]
+
+ DONE  Compiled successfully in 2679ms                                                                                                                                19:17:27
+
+  File                            Size                                                                 Gzipped
+
+  ../views/js/chunk-vendors.js    121.79 KiB                                                           44.16 KiB
+  ../views/js/app.js              5.41 KiB                                                             1.94 KiB
+
+  Images and other types of assets omitted.
+
+ DONE  Build complete. The ../views directory is ready to be deployed.
+ INFO  Check out deployment instructions at https://cli.vuejs.org/guide/deployment.html
+```
+
+## 4- Display of the application
+
+The application is now ready to be integrated inside a page of PrestaShop.
+There are several steps to complete:
+
+* **Adding a div with the proper ID inside a template**
+
+*modules/[module_name]/views/templates/admin/app.tpl*
+
+```tpl
+<link href="{$pathApp|escape:'htmlall':'UTF-8'}" rel=preload as=script>
```

Comment:
@Quetzacoalt91 is `escape` still needed for back-office in 1.7?

Reply:
That's what I thought, I might be not up to date

Reply:
Escape has been enabled everywhere with https://github.com/PrestaShop/PrestaShop/pull/4176, but hen only enable for the front office with https://github.com/PrestaShop/PrestaShop/commit/f2fc2f791782988cf97de7ba58141f02a9340ab8.

I did not see other changes and `$smarty->escape_html` is still `false` by default. That's why I consider that we still need to escape variables on the back office.

Reply:
@Quetzacoalt91 I think you're right 👍